src/main/java/io/excaliburfrc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,47 @@
+package io.excaliburfrc.robot.subsystems;
+
+import static io.excaliburfrc.robot.Constants.VisionConstants.*;
+
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonUtils;
+import org.photonvision.targeting.PhotonPipelineResult;
+
+public class Vision extends SubsystemBase {
+  private final PhotonCamera camera = new PhotonCamera("limelight");
+  private PhotonPipelineResult result = camera.getLatestResult();
+
+  public double getRotation() {
+    if (!result.hasTargets()) {
+      return 0;
+    }
+    return result.getBestTarget().getYaw();
+  }
+
+  public double getDistance() {
+    if (!result.hasTargets()) {
+      return 0;
+    }
+
+    return PhotonUtils.calculateDistanceToTargetMeters(
+        CAMERA_HEIGHT,
+        TARGET_HEIGHT,
+        CAMERA_PITCH,
+        Units.degreesToRadians(result.getBestTarget().getPitch()));
+  }
+
+  public Transform2d getTransform() {
+    if (!result.hasTargets()) {
+      return new Transform2d();
+    }
```

Comment:
I think there's probably a better way to handle no targets. Maybe `Optional`?
We should probably discuss this, I'm not sure that `Optional` is a good option either